www/js/theme_editor.js

```diff
@@ -19,6 +20,83 @@ const themesWithCustomBg = themes
                             .filter(theme => theme.type === 'COLOR' || theme.type === 'SPECIAL')
                             .map(theme => theme.key);
 
+/*
+ * Helper Functions
+ */
+
+const uploadErrorNotification = (message) => {
+  new Noty({
+    type: 'error',
+    text: message,
+    timeout: 3000,
+    modal: true,
+  }).show();
+};
+
+const imageCheck = (filePath) => {
+  const acceptedExtensions = ['.png', '.PNG', '.jpg', '.JPG', '.jpeg', '.JPEG'];
+  const extension = path.extname(filePath);
+
+  return acceptedExtensions.includes(extension);
+};
+
+/*
+ * DOM Factories
+ */
+
+const recentSwatchFactory = backgroundPath =>
+  h(
+    'li.theme-instance.custom-bg',
+    {
+      style: {
+        'background-image': `url(${backgroundPath.replace(/(\s)/g, '\\ ')})`,
+      },
+      onclick: () => {
+        store.setUserPref('app.theme', themesWithCustomBg[0]);
+        store.setUserPref('app.themebg', {
+          type: 'custom',
+          url: backgroundPath.replace(/(\s)/g, '\\ '),
+        });
+        global.core.platformMethod('updateSettings');
+      },
+    },
+    h(
+      'button.delete-btn',
+      {
+        onclick: (evt) => {
+          evt.stopPropagation();
+          fs.unlink(backgroundPath, (error) => {
+            if (error) {
+              uploadErrorNotification(`Unable to delete that image. Press cmd+r / ctrl+r to refresh. - ${error}`);
+            } else {
+              evt.target.parentNode.parentNode.classList.add('delete-animate');
+            }
+          });
+        },
+      },
+      h('i.fa.fa-trash-o'),
+    ),
+  );
+
+
+const upsertCustomBackgrounds = (themesContainer) => {
+  document.querySelectorAll('.custom-bg').forEach((swatch) => {
```

Comment:
anyway to wrap the custom bgs in a container and remove at one time instead of having to loop?